client/app/scripts/superdesk-highlights/highlights.js

```diff
@@ -250,23 +250,50 @@
         };
     }
 
-    PackageHighlightsDropdownDirective.$inject = ['superdesk', 'desks', 'highlightsService'];
-    function PackageHighlightsDropdownDirective(superdesk, desks, highlightsService) {
+    PackageHighlightsDropdownDirective.$inject = ['superdesk', 'desks', 'highlightsService', '$location', '$route'];
+    function PackageHighlightsDropdownDirective(superdesk, desks, highlightsService, $location, $route) {
         return {
             templateUrl: 'scripts/superdesk-highlights/views/package_highlights_dropdown_directive.html',
             link: function(scope) {
+                scope.$watch(function() {
+                    return desks.active;
+                }, function(active) {
+                    scope.selected = active;
+                    highlightsService.get(desks.getCurrentDeskId()).then(function(result) {
+                        scope.highlights = result._items;
+                        scope.hasHighlights = _.size(scope.highlights) > 0;
+                    });
+                });
+
+                scope.listHighlight = function(highlight) {
+                    var data = {highlightId: highlight._id, name: highlight.name};
+                    $location.url('workspace/highlights-view?data=' + encodeURIComponent(JSON.stringify(data)));
```

Comment:
it would be easier to use this from other places if the url is like `workspace/highlights/_id` or `workspace/highlights?highlight=<id>`, name can be fetched in activity controller.